src/main/java/duke/FileProcessor.java

```diff
@@ -0,0 +1,73 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.*;
+import java.util.ArrayList;
+
+public class FileProcessor {
+
+    public File dukeTextFile;
+
+    public FileProcessor(ArrayList<Task> taskList) throws IOException {
+        this.dukeTextFile = new File("duke.txt");
+        try {
```

Comment:
To improve readability, you may want to avoid deep nesting. Perhaps consider refactoring the deeply nested part to a method. [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting)